core/crypto/crypto_store/crypto_store_impl.cpp

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/crypto_store/crypto_store_impl.hpp"
+
+#include <fstream>
+
+#include <gsl/span>
+#include "common/visitor.hpp"
+#include "crypto/bip39/mnemonic.hpp"
+
+namespace kagome::crypto {
+  using DirectoryIterator = std::filesystem::directory_iterator;
+
+  struct KeyFileInfo {
+    KeyTypeId key_type_id;
+    store::PublicKey public_key;
+  };
+
+  using MnemonicPhrase = std::string;
+  using KeyFileContent = boost::variant<store::Seed, MnemonicPhrase>;
+
+  namespace {
+    template <class T>
+    outcome::result<KeyTypeId> keyTypeFromBytes(gsl::span<const T> bytes) {
```

Comment:
the only place where you use this function is 1 line below where you create key_type_str of length 4. So why bother adding these additional checks and returning result? Probably it makes sense to just return KeyTypeId and add boost assert to check the size

Reply:
done